### PR TITLE
Return valid fileinfo objects for part files

### PR DESCRIPTION
**Pull request by icewind1991**
Also makes the Node api work for part files

see https://github.com/owncloud/core/pull/13275#issuecomment-69649348

cc @DeepDiver1975 @PVince81

### CODE REVIEW

File: lib/private/files/view.php [hunk 1]
```diff
@@ -1308,4 +1311,32 @@ private function isTargetAllowed($target) {
...
+			$storage,
+			$internalPath,
+			[
+				'fileid' => null,
+				'mimetype' => $storage->getMimeType($internalPath),
+				'name' => basename($path),
+				'etag' => null,
+				'size' => $storage->filesize($internalPath),
+				'mtime' => $storage->filemtime($internalPath),
+				'encrypted' => false,
```
Comment (PVince81) [inline lib/private/files/view.php]: Part files are usually encrypted when encryption is enabled, so not sure about this value here.

Comment (PVince81) [inline lib/private/files/view.php]: CC @schiesbn

File: lib/private/files/view.php [hunk 2]
```diff
@@ -1308,4 +1311,32 @@ private function isTargetAllowed($target) {
...
+	private function getPartFileInfo($path) {
+		$mount  = $this->getMount($path);
+		$storage = $mount->getStorage();
+		$internalPath = $mount->getInternalPath($this->getAbsolutePath($path));
+		return new FileInfo(
+			$this->getAbsolutePath($path),
+			$storage,
+			$internalPath,
+			[
+				'fileid' => null,
```
Comment (DeepDiver1975) [inline lib/private/files/view.php]: I'm not too sure about this - e.g. search_lucene is operating on the fileId @butonic

**Comment by DeepDiver1975** [comment]
tested :+1: THX

**Comment by DeepDiver1975** [comment]
also fixes https://github.com/owncloud/core/issues/13242

File: lib/private/files/view.php (hunk 2 above)
Comment (icewind1991) [inline lib/private/files/view.php]: That will have to be handled in the app using it, cant give a valid id here

**Comment by DeepDiver1975** [comment]
@PVince81 @MorrisJobke @derkostka  - you might want to test this with respect to #13242 - THX

**Comment by derkostka** [comment]
:+1: , fixes #13242.

**Comment by derkostka** [comment]
@DeepDiver1975 @icewind1991 : Can you Backport this to stable7 ? I tested the issue i had on an 7.0.4 instance, seems also to happen there (note i do not have the music app enabled in this instance, but i use the same android device to upload files)

**Comment by icewind1991** [comment]
Should be safe to backport

@karlitschek ?

**Comment by karlitschek** [comment]
backport is fine :+1:

File: lib/private/files/view.php (hunk 2 above)
Comment (DeepDiver1975) [inline lib/private/files/view.php]: sure

**Comment by DeepDiver1975** [comment]
summoning @MorrisJobke regarding backport and proper label-ing :speak_no_evil: THX

**Comment by MorrisJobke** [comment]
stable7 0f04bfc

**Comment by DeepDiver1975** [comment]
thx @MorrisJobke

**Pull request merged**